File: nqueen.cpp
#include<bits/stdc++.h>
using namespace std;

#define n 8

void printSol(int board[][n]){
    for(int i=0; i<n; i++){
        for(int j=0; j<n; j++){
            cout<<board[i][j]<<' ';
        }
        cout<<endl;
    }
}

bool isSafe(int board[][n], int row, int col){
    for(int i=n; i>=0; i--){
        if(board[i][col] == 1)
         return false;
    }
    for(int i=row, j=col; i>=0 and j>=0; i--, j--){
        if(board[i][j] == 1)
            return false;
    }
    for(int i=row,j=col; i>=0 and j<n; i--, j++){
        if(board[i][j] == 1)
            return false;
    }
    return true;
}

File: nqueen.py
#help in creating nqueen in python 
import os
import time

def printBoard(board, n):
    string = "-----"*n
    for i in range(0, n):
        if(i==0):
            print(string)
        for j in range(0, n):
            if(j==0):
                print("|", end=" ")
            print(board[i][j], end=" | ")
        print()
        # print("---------------------------------")
        print(string)


def isSafe(board, row, col, n):
    for r in range(0, row):
        if(board[r][col] == 1):
            return False
    
    r=row
    c=col
    while(r>=0 and c>=0):
        if(board[r][c] == 1):
            return False
        r-=1
        c-=1
    r=row
    c=col
    while(r>=0 and c<n):
        if board[r][c] == 1:
            return False
        r-=1
        c+=1

    return True

def solve(board, row, n):
    if(row>=n):
        return True

    for col in range(0, n):
        # printBoard(board, n)
        # time.sleep(0.01)
        # os.system("clear")
        if(isSafe(board, row, col , n)):
            board[row][col] = 1
            if(solve(board, row+1, n)):
                return True
            board[row][col] = 0

            def main():
    n = int(input("Enter The Number of Rows "))
    board = [[0 for i in range(0, n)]for j in range(0, n)]
    
    # printBoard(board, n)


    if(solve(board, 0, n)):
        printBoard(board, n)
    
    else:
        print("Can Not Be Solved")
        printBoard(board, n)
